docs/source/TestParameters.rst

```diff
@@ -0,0 +1,945 @@
+   in `Yaml_to_mux plugin`_.
+
+.. warning:: The multiplexer is under heavy refactor and some of the APIs
+   might still change in the following months (written on 2016-01-22),
+   then we'll do our best to keep the public interfaces as stable as
+   possible.
+
+Avocado allows passing parameters to each executed test or even running
+several variants of each test. These parameters are available in (test's)
+``self.params`` and are of :mod:`avocado.core.varianter.AvocadoParams` type.
+
+The data for ``self.params`` are supplied by
+:mod:`avocado.core.varianter.Varianter` which asks all registered plugins
+for variants or uses default when no variants are defined.
+
+Overall picture of how the params handling works is:
+
+.. following figure is not really a C code, but it renders well and it
+   increases the visibility.
+
+.. code-block:: c
+
+       +-----------+
+       |           |  // Test uses variant to produce AvocadoParams
+       |   Test    |
+       |           |
+       +-----^-----+
+             |  // single variant is passed to Test
+             |
+       +-----------+
+       |  Runner   |  // iterates through tests and runs each test with
+       +-----^-----+  // all variants supplied by Varianter
+             |
+             |
+   +-------------------+ provide variants +-----------------------+
+   |                   |<-----------------|                       |
+   | Varianter API     |                  | Varianter plugins API |
+   |                   |----------------->|                       |
+   +-------------------+  update defaults +-----------------------+
+             ^                                ^
+             |                                |
+             |  // default params injected    |  // All plugins are invoked
+   +--------------------------------------+   |  // in turns
+   | +--------------+ +-----------------+ |   |
+   | | avocado-virt | | other providers | |   |
+   | +--------------+ +-----------------+ |   |
+   +--------------------------------------+   |
+                                              |
+                 +----------------------------+-----+
+                 |                                  |
+                 |                                  |
+                 v                                  v
+       +--------------------+           +-------------------------+
+       | yaml_to_mux plugin |           | Other variant plugin(s) |
+       +-----^--------------+           +-------------------------+
+             |
+             |  // yaml is parsed to MuxTree,
+             |  // multiplexed and yields variants
+       +---------------------------------+
+       | +------------+ +--------------+ |
+       | | --mux-yaml | | --mux-inject | |
+       | +------------+ +--------------+ |
+       +---------------------------------+
+
+
+Let's introduce the basic keywords.
+
+Test's default params
+~~~~~~~~~~~~~~~~~~~~~
+
+:mod:`avocado.core.test.Test.default_params`
```

Comment:
It's minor, but yes, a `FIXME` (or `NOTE`) is useful to at least remind us of what's happening there.  I imagine that if I encounter this again 2 months from now, I'd try to fix it, and probably waste a couple of hours because seems to be an issue with sphinx.